pkg/odo/cli/component/create.go

```diff
@@ -759,9 +759,21 @@ func (co *CreateOptions) downloadProject() error {
 		return errors.Errorf("Project type not supported")
 	}
 
-	err = util.GetAndExtractZip(zipUrl, path)
-	if err != nil {
-		return err
+	if project.Source.SparseCheckoutDir != nil && *project.Source.SparseCheckoutDir != "" {
+
+		sparseCheckoutDir := *project.Source.SparseCheckoutDir
+		err = util.GetAndExtractZip(zipUrl, path, sparseCheckoutDir)
+		if err != nil {
+			return err
```

Comment:
Can you wrap the errors that are returned in this function with `errors.Wrap` / `errors.Wrapf` with a message?